37.md

```diff
@@ -0,0 +1,95 @@
+NIP-37
+======
+
+Methods for dealing with lost or compromised keys
+-------------------------------------------------
+
+`draft` `optional` `author:fiatjaf`
+
+This NIP defines a series of methods that can be used to make catastrophic key loss events less horrible.
+
+The intention here is not to claim that there is a generalized "key rotation" scheme on Nostr, but still make it
+so that in most cases people are able to move to new keys and not lost 100% of their followers and reputation
+when that happens, and also that it is possible to prevent most of the reputational damage that could be
+caused by having an evil hacker take control of someone else's keys.
+
+The implementation of this NIP could be done directly in social clients or in dedicated clients that users would
+visit from time to time in order to have an automated scan be done on their contacts and then the dedicated app
+could suggest migrating from old keys to new keys, or suggest stop following compromised keys, or even perform
+these migrations automatically.
+
+### Method A: Simple Key Deletion
+
+Based on https://github.com/nostr-protocol/nips/pull/377, the idea is that someone, after noticing that their key
+has been compromised and deciding that it is not possible to use any of the other methods here described, can
+proceed to publish the following event:
+
+```js
+{
+  "kind": 10529,
+  "pubkey": "<compromised-pubkey>",
+  "tags": [
```

Comment:
One more tag could be the list of pubkeys actually revoked where the signing pubkey is an authoritative key where the subkeys have previously signed the authoritative key to give a compromise revocation mandate to the authoritative one. Though I think this more method C suggested at the end.